### PR TITLE
Fix/rewrite outbox tests

### PR DESCRIPTION
**Pull request by jonasdmoeller**
Refactored OutboxProcessorEndpoint integration tests to not use function host, but instead mock calls to DomainEventPublisher.

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/ChargeInformation/OutboxMessageProcessorEndpointTests.cs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                var outboxMessage = await CreateStoredOutboxMessage(domainEvent);
+
+                var outboxProcessor = new OutboxMessageProcessorEndpoint(
+                    _fixture.GetService<IOutboxMessageRepository>(),
+                    _fixture.GetService<IOutboxMessageParser>(),
+                    _fixture.GetService<IClock>(),
+                    _correlationContext,
+                    domainEventDispatcher.Object,
+                    _fixture.UnitOfWork,
+                    _fixture.GetService<ILoggerFactory>());
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/ChargeInformation/OutboxMessageProcessorEndpointTests.cs:74]: The `sut` has the same dependencies for tests. I prefer we put this part in a method for reuse in all test-methods.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/Charges/OutboxMessageProcessorEndpointTests.cs [hunk 2]
```diff
@@ -1,222 +0,0 @@
...
-                        It.IsAny<CancellationToken>()))
-                    .Callback<DomainEvent, CancellationToken>((_, _) => { });
-                await sut.RunAsync(timerInfo);
-
-                outboxMessage = chargesDatabaseReadContext.OutboxMessages.Single(x => x.Id == outboxMessage.Id);
-                outboxMessage.ProcessedDate.Should().Be(now);
-            }
-
-            [Fact]
-            public async Task GivenNewRejectedEvent_WhenRunAsync_ThenRejectedEventIsProcessed()
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/Charges/OutboxMessageProcessorEndpointTests.cs:184]: No test method corresponding to this test is implemented in the new `OutboxMessageProcessorEndpointTests.cs`. Is this test considered redundant?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Messaging/IDomainEventDispatcher.cs [hunk 3]
```diff
@@ -26,7 +25,6 @@ public interface IDomainEventDispatcher
         /// Dispatch domain event
         /// </summary>
         /// <param name="domainEvent"></param>
-        /// <param name="cancellationToken"></param>
-        Task DispatchAsync<T>(T domainEvent, CancellationToken cancellationToken = default);
+        Task DispatchAsync<T>(T domainEvent);
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Messaging/IDomainEventDispatcher.cs:28]: Removal of the `CancellationToken` is sneaked in, and, having read about it, I agree, we do not need it, at least not for now 👍

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/Charges/OutboxMessageProcessorEndpointTests.cs (hunk 2 above)
Comment (jonasdmoeller) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/Charges/OutboxMessageProcessorEndpointTests.cs:184]: Yes, I found it redundant with the `RunAsync_WhenRejectedOutboxMessageIsRead_ThenDomainEventRaised_AndProcessedDateIsSet` test case

**Pull request merged**